catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/FormulaEditorEditTextTest.java

```diff
@@ -0,0 +1,407 @@
+		onFormulaEditor()
+				.performEnterFormula("(1.3)");
+		onFormulaEditor()
+				.checkShows("( 1" + decimalMark + "3 ) ");
+
+		setAbsoluteCursorPosition(1);
+		onView(withId(R.id.formula_editor_keyboard_delete))
+				.perform(click());
+		setAbsoluteCursorPosition(7);
+		onView(withId(R.id.formula_editor_keyboard_delete))
+				.perform(click());
+		onFormulaEditor()
+				.checkShows("1" + decimalMark + "3 ");
+	}
+
+	@Category({Cat.CatrobatLanguage.class, Level.Smoke.class})
+	@Test
+	public void testLongClickDeletion() {
+		onFormulaEditor()
+				.performEnterFormula("1234");
+		onView(withId(R.id.formula_editor_keyboard_delete))
+				.perform(longClick());
+		onFormulaEditor()
+				.checkShows(" ");
+		onFormulaEditor()
+				.performEnterFormula("1234");
+		setAbsoluteCursorPosition(2);
+		onView(withId(R.id.formula_editor_keyboard_delete))
+				.perform(longClick());
+		onView(withId(R.id.formula_editor_edit_field))
+				.check(matches(withText("34 ")));
+	}
+
+	@Category({Cat.CatrobatLanguage.class, Level.Smoke.class})
+	@Test
+	public void testFunctionDeletion() {
+		onView(withId(R.id.formula_editor_keyboard_function))
+				.perform(click());
+		onView(withText(containsString(random)))
+				.perform(click());
+		onView(withId(R.id.formula_editor_keyboard_delete))
+				.perform(click());
+		onView(withId(R.id.formula_editor_keyboard_delete))
+				.perform(click());
+
+		onView(withId(R.id.formula_editor_keyboard_function))
+				.perform(click());
+
+		onView(allOf(withText(containsString(random)), withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)))
+				.perform(click());
+
+		onView(withId(R.id.formula_editor_keyboard_delete))
+				.perform(click());
+		onView(withId(R.id.formula_editor_keyboard_delete))
+				.perform(click());
+		onFormulaEditor()
+				.checkShows(" ");
+	}
+
+	@Category({Cat.CatrobatLanguage.class, Level.Smoke.class})
+	@Test
+	public void testNumberInsertion() {
+
+		onFormulaEditor()
+				.performEnterFormula(".1");
+		onFormulaEditor()
+				.checkShows("0" + decimalMark + "1 ");
+		setAbsoluteCursorPosition(0);
+		onFormulaEditor()
+				.performEnterFormula("5");
+		setAbsoluteCursorPosition(3);
+		onFormulaEditor()
+				.performEnterFormula("6");
+		onFormulaEditor()
+				.checkShows("50" + decimalMark + "61 ");
+	}
+
+	@Category({Cat.CatrobatLanguage.class, Level.Smoke.class})
+	@Test
+	public void testGoBackToDiscardChanges() {
+
+		onFormulaEditor()
+				.performEnterFormula("99.9+");
+		pressBack();
+		onView(withText(getCurrentActivity().getString(R.string.formula_editor_discard_changes_dialog_title)))
+				.check(matches(isDisplayed()));
+		onView(withText(no))
+				.perform(click());
+		checkToastMessage(R.string.formula_editor_changes_discarded);
+
+		onBrickAtPosition(1)
+				.checkShowsText("0 ");
+	}
+
+	@Category({Cat.CatrobatLanguage.class, Level.Smoke.class})
+	@Test
+	public void testDiscardDialog() {
+		onFormulaEditor()
+				.performEnterFormula("99");
+		pressBack();
```

Comment:
please split this test into smaller tests. its really hard to follow whats going on.